### PR TITLE
PYTHON-4558 Fix coverage and XML results for async tests

### PR DESCRIPTION
**Pull request by ShaneHarvey**
https://jira.mongodb.org/browse/PYTHON-4558

### CODE REVIEW

File: .evergreen/run-tests.sh [hunk 1]
```diff
@@ -257,8 +254,7 @@ PIP_QUIET=0 python -m pip list
 if [ -z "$GREEN_FRAMEWORK" ]; then
     # Use --capture=tee-sys so pytest prints test output inline:
     # https://docs.pytest.org/en/stable/how-to/capture-stdout-stderr.html
-    python -m pytest -v --capture=tee-sys --durations=5 --maxfail=10 $TEST_ARGS
-    python -m pytest -v --capture=tee-sys --durations=5 --maxfail=10 test/asynchronous/ $TEST_ARGS
+    python -m pytest -v --capture=tee-sys --durations=5 --maxfail=10 $TEST_SUITES $TEST_ARGS
```
Comment (NoahStapp) [inline .evergreen/run-tests.sh:257]: Nice, this is a clean solution. Do we still need `$TEST_ARGS`?

File: .github/workflows/test-python.yml [hunk 2]
```diff
@@ -205,12 +205,11 @@ jobs:
...
+        - name: Run connect test from sdist
           shell: bash
           run: |
             cd sdist/test
             ls
             which python
             pip install -e ".[test]"
-            pytest -v
-            pytest -v test/asynchronous/
+            PYMONGO_MUST_CONNECT=1 pytest -v test/test_client_context.py
```
Comment (NoahStapp) [inline .github/workflows/test-python.yml:215]: Has this test been just running the wrong suite this whole time?

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: .github/workflows/test-python.yml (hunk 2 above)
Comment (ShaneHarvey) [reply .github/workflows/test-python.yml:215]: No but it just seems unnessecary to run the whole suite from the sdist, we already run the whole suite elsewhere on the same commit.

File: .evergreen/run-tests.sh (hunk 1 above)
Comment (ShaneHarvey) [reply .evergreen/run-tests.sh:257]: TEST_ARGS is still used to add `--cov`.

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**